en/administration/cluster/loadbalancer/haproxy.md

```diff
@@ -6,155 +6,43 @@ http://support.rundeck.com/customer/en/portal/articles/2894842-health-check-base
-
-The example below uses a python script to demonstrate this.
-
-Create the following script on `$RDECK_BASE/server/exp/webapp/WEB-INF/cgi-bin/status`
- 
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ {.python}
-#!/usr/bin/python
-
-import sys
-import os
-import cgi
-
-from cgi import escape
-import requests
-import json
-
-form = cgi.FieldStorage()
-token = form.getvalue("token")
-executionMode= form.getvalue("executionMode")
-
-url = 'http://localhost:4440/api/20/system/info?format=json&authtoken=' + token
-
-response = requests.post(url)
-info = json.loads(response.text)
-currentExecutionMode = info["system"]["executions"]["executionMode"]
-
-if currentExecutionMode == executionMode:
-    status = "200"
-else:
-    status = "403"
-
-
-print "Content-type: text/html"
-print 'Status:' + status + ' '
-
-print
-print "<!doctype html>"
-print "<html>"
-print "<body>"
-
-print 'Current Execution Mode:' + currentExecutionMode
-print "<br>"
-print 'Expected Execution Mode:' + executionMode
-print "</body>"
-print "</html>"
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-After the script has been deployed use your browser to visit the new health check page.
-
-The URL accepts a parameter called "executionMode" which takes one of two values: active or passive. If specified mode matches the actual mode the page will return HTTP response code 200 otherwise it will return 403.
-
-For example: `http://localhost:4440/cgi-bin/status?token=<token>&executionMode=active`
-
-![Example health check](../../../figures/haproxy-health-check.png)
+backend default_service
+mode http
+balance roundrobin
+cookie JSESSIONID prefix nocache
+server rundeck1 192.168.0.1:4440 cookie rundeck1
+server rundeck2 192.168.0.2:4440 cookie rundeck2
+server rundeck3 192.168.0.3:4440 cookie rundeck3
+```
 
-### On HAProxy, add the following settings
+### Using healtcheck 
 
-The web load balancer serving traffic to the Rundeck instance should be configured with the health check.
+To avoid sending traffic to unavailable nodes, HAProxy can check the health of
+their downstream services via http or tcp. 
 
-* Add the "httpchk" with the URI of the health check. The token value and the execution mode that you want to use are needed.
+* Add the "httpchk" with the URI of the health check (Rundeck API "system info" endpoint). 
+  The token value and the execution mode that you want to use are needed.
 * Add the "http-check" with the expected status (in this case 200)
  
 ```
 backend default_service
+mode http
+balance roundrobin
 cookie JSESSIONID prefix nocache
-option httpchk get /cgi-bin/status?token=<TOKEN_VALUE>&executionMode=active
+option httpchk get /api/23/system/info?authtoken=<TOKEN_VALUE>
```

Comment:
Oh nice, so the health check script is built into rundeck now?

Reply:
Hey Carlo, 

No, the example that we have on the docs was creating a "custom" heartbeat for HAPROXY using CGI script on rundeck. However, that is not possible anymore on rundeck 3 so I removed. I replaced with an example using a heartbeat with the API (call system/info).

Reply:
rundeck 3 will have api access to the metrics info, including /metrics/ping which would be a solution here.  see https://github.com/rundeck/docs/blob/3.0.x/en/api/rundeck-api.md#metrics-ping